ios/FluentUI/Vnext/PersonaGrid/PersonaGridItem.swift

```diff
@@ -0,0 +1,146 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import SwiftUI
+
+/// `MSFPersonaGridItemState` contains PersonaGridItem properties in addition to a subset of the MSFAvatarState protocol.
+///
+/// `onTapAction` provides tap gesture for PersonaGridItem.
+///
+@objc public protocol MSFPersonaGridItemState {
+    var image: UIImage? { get set }
+    var primaryText: String? { get set }
+    var secondaryText: String? { get set }
+    var size: MSFAvatarSize { get set }
+
+    var onTapAction: (() -> Void)? { get set }
+}
+
+/// Properties that make up PersonaGridItem content
+class MSFPersonaGridItemViewStateImpl: NSObject, ObservableObject, Identifiable, MSFPersonaGridItemState {
+    public var onTapAction: (() -> Void)?
+
+    @ObservedObject private var tokens: MSFPersonaGridItemTokens
+    private var avatarState: MSFAvatarState
+
+    var image: UIImage? {
+        get {
+            return avatarState.image
+        }
+
+        set {
+            avatarState.image = newValue
+        }
+    }
+
+    var primaryText: String? {
+        get {
+            return avatarState.primaryText
+        }
+
+        set {
+            avatarState.primaryText = newValue
+        }
+    }
+
+    var secondaryText: String? {
+        get {
+            return avatarState.secondaryText
+        }
+
+        set {
+            avatarState.secondaryText = newValue
+        }
+    }
+
+    var size: MSFAvatarSize {
```

Comment:
It will trigger a re-layout, but it won't currently trigger the text appearing or disappearing. I'd really rather hide all this stuff away from the public interface, but I can only think of three options:
1. We implement the full `MSFAvatarState` protocol, per https://github.com/microsoft/fluentui-apple/pull/597#discussion_r648612510 (and thus make this property public)
2. We manually specify which properties from `Avatar` we want to make public, in which case we risk having API drift if new props are added to `Avatar`.
3. We split the `MSFAvatarState` protocol into two: one for styling and one for... sizing, I guess?

Option 3 is cleanest for `Persona`, but feels kinda arbitrary. The other two have their own drawbacks. What are your thoughts?

Reply:
I'm leaning towards defining all the properties that we want to expose in the MSFPersonaBadgeState and forwarding what we need to forward in the implementation to the avatar state.

One of the reasons for that is when we need for example to have the "Signed Out" state of the individual use of the persona badge:
The calling code may want to display "Sign In" in the main label, and the Avatar can't display the "S" initials. It will instead need to be set to a specific avatar style and both primary and secondary text properties need to be set to an empty value. This would be more a Persona Badge implementation detail that we can do in its state object instead of just exposing all the Avatar state which would make it harder for the calling code to implement this scenario on its own.

We don't have to implement this functionality just yet, but I think the control will be ready for that enhancement if we don't just expose the AvatarState directly.

The other benefit is the one we already discussed about the size. We can just choose not to expose the Avatar size and control it internally when the PersonaBadge size is set.

What do you think?

Reply:
Makes perfect sense to me! That's actually what I had in my first iteration; it was just a much smaller subset of properties. I'll scrub through and see what makes sense in this new more-expanded world.